File: tests/indexing/parsing/lang/fixtures/go/BodySizes.go
package example

// externalFunction is implemented in assembly elsewhere, so it has no body.
func externalFunction(a, b int) int

func emptyBody() {}

func withLogic(items []int) (int, error) {
	if len(items) == 0 {
		return 0, nil
	}
	max := items[0]
	for _, item := range items[1:] {
		if item > max {
			max = item
		}
	}
	return max, nil
}

File: tests/indexing/parsing/lang/fixtures/rust/BodySizes.rs
pub trait Shape {
    fn external_function(&self) -> f64;
}

pub fn empty_body() {}

pub fn with_logic(numbers: &[i32]) -> i32 {
    let mut total = 0;
    for n in numbers {
        if n % 2 == 0 {
            total += n;
        } else {
            total -= n;
        }
    }
    total
}
